storm-core/src/clj/org/apache/storm/ui/core.clj

```diff
@@ -466,6 +466,21 @@
        "assignedCpu" (.get_assigned_cpu t)})
     "schedulerDisplayResource" (*STORM-CONF* Config/SCHEDULER_DISPLAY_RESOURCE)}))
 
+(defn get-topology-id [topology-name-or-id]
+  (let [summary ((all-topologies-summary) "topologies")
+        filter-fun-name (fn[topo-summary] (= (topo-summary "name") topology-name-or-id))
+        filter-fun-id (fn[topo-summary] (= (topo-summary "id") topology-name-or-id))
+        matching-topologies-by-name (filter filter-fun-name summary)
+        matching-topologies-by-id (filter filter-fun-id summary)
+        matching-topologies
+        (cond
+          (not-empty matching-topologies-by-name) matching-topologies-by-name
+          (not-empty matching-topologies-by-id) matching-topologies-by-id
+          :else nil)
+        _ (when (empty? matching-topologies) (throw (NotAliveException. (str topology-name-or-id " is not alive"))))
+        id ((first matching-topologies) "id")]
+    id))
```

Comment:
@knusbaum Thats true. But if we prioritize ids would not your topology become inaccessible? I think going forward we should use topology names in ui and elsewhere so that we can have permanent links and since thats the identifier given by the user. Support for lookup with ids was added for backwards compatibility as suggested by @revans2 and others.